test/jdk/tools/jlink/basic/AllModulePath.java

```diff
@@ -84,60 +101,135 @@ public void setup() throws Throwable {
+        if (isExplodedJDKImage()) {
             return;
         }
 
-        // create custom image
-        Path image = Paths.get("image");
-        createImage(image, "--add-modules", "ALL-MODULE-PATH");
+        Path image = HELPER.createNewImageDir("image");
+        List<String> opts = List.of("--module-path", MODS.toString(),
+                                    "--output", image.toString(),
+                                    "--add-modules", "ALL-MODULE-PATH");
+        createImage(image, opts, true /* success */);
 
         Set<String> modules = new HashSet<>();
-        Files.find(JMODS, 1, (Path p, BasicFileAttributes attr) ->
-                                p.toString().endsWith(".jmod"))
-             .map(p -> JMODS.relativize(p).toString())
-             .map(n -> n.substring(0, n.length()-5))
-             .forEach(modules::add);
+        // java.base is a dependency of any external module
+        modules.add("java.base");
         modules.add("m1");
         modules.add("test");
         checkModules(image, modules);
     }
 
+    /*
+     * Since ALL-MODULE-PATH does not allow --limit-modules. Add a test that
+     * includes just a single module from the module path. This is just for
+     * completeness and shows the intended replacement.
+     */
     @Test
-    public void testLimitModules() throws Throwable {
-        if (Files.notExists(JMODS)) {
+    public void testSubsetModules() throws Throwable {
+        if (isExplodedJDKImage()) {
             return;
         }
 
         // create custom image
-        Path image = Paths.get("image1");
-        createImage(image,
-                    "--add-modules", "ALL-MODULE-PATH",
-                    "--limit-modules", "m1");
+        Path image = HELPER.createNewImageDir("image1");
+        // Instead of --add-modules ALL-MODULE-PATH [...] --limit-modules m1 do:
+        List<String> opts = List.of("--module-path", MODS.toString(),
+                                    "--output", image.toString(),
+                                    "--add-modules", "m1");
+        createImage(image, opts, true /* success */);
 
         checkModules(image, Set.of("m1", "java.base"));
     }
 
+    /*
+     * --add-modules *includes* ALL-MODULE-PATH with an existing module path
+     */
     @Test
     public void testAddModules() throws Throwable {
-        if (Files.notExists(JMODS)) {
+        if (isExplodedJDKImage()) {
             return;
         }
 
         // create custom image
-        Path image = Paths.get("image2");
-        createImage(image,
-                    "--add-modules", "m1,test",
-                    "--add-modules", "ALL-MODULE-PATH",
-                    "--limit-modules", "java.base");
+        Path image = HELPER.createNewImageDir("image2");
+        List<String> opts = List.of("--module-path", MODS.toString(),
+                                    "--output", image.toString(),
+                                    "--add-modules", "m1,test",
```

Comment:
```suggestion
                                    "--add-modules", "m1",
```

This can show clearly that `ALL-MODULE-PATH` includes all modules on the module path.